src/envs/behavior.py

```diff
@@ -244,56 +308,104 @@ def _classifier(s: State, o: Sequence[Object]) -> bool:
+        assert state.allclose(self._current_ig_state_to_state())
+        assert len(objs) == 2
+        ig_obj = self.object_to_ig_object(objs[0])
+        ig_other_obj = self.object_to_ig_object(objs[1])
+        return (np.linalg.norm(  # type: ignore
+            np.array(ig_obj.get_position()) -
+            np.array(ig_other_obj.get_position())) < 2)
+
+    def _reachable_nothing_classifier(self, state: State,\
+        objs: Sequence[Object]) -> bool:
+        # Check allclose() here for uniformity with _create_classifier_from_bddl
+        assert state.allclose(self._current_ig_state_to_state())
+        assert len(objs) == 1
+        for obj in state:
+            if self._reachable_classifier(
+                    state=state, objs=[obj, objs[0]]) and (obj != objs[0]):
+                return False
+        return True
+
     def _get_grasped_objects(self, state: State) -> Set[Object]:
         grasped_objs = set()
         for obj in state:
-            ig_obj = self._object_to_ig_object(obj)
-            if any(self._env.robots[0].is_grasping(ig_obj)):
+            ig_obj = self.object_to_ig_object(obj)
+
+            # NOTE: The below block is necessary because somehow the body_id
+            # is sometimes a 1-element list...
+            if isinstance(ig_obj.body_id, list):
+                assert len(ig_obj.body_id) == 1
+                ig_obj.body_id = ig_obj.body_id[0]
+
+            if any(self.behavior_env.robots[0].is_grasping(ig_obj.body_id)):
                 grasped_objs.add(obj)
+
         return grasped_objs
 
-    def _handempty_classifier(self, state: State,
-                              objs: Sequence[Object]) -> bool:
-        # Check allclose() here for uniformity with _create_classifier_from_bddl
+    def _handempty_classifier(self, state: State,\
+        objs: Sequence[Object]) -> bool:
+        # Check allclose() here for uniformity with
+        # _create_classifier_from_bddl
         assert state.allclose(self._current_ig_state_to_state())
         assert len(objs) == 0
         grasped_objs = self._get_grasped_objects(state)
         return len(grasped_objs) == 0
 
-    def _holding_classifier(self, state: State,
-                            objs: Sequence[Object]) -> bool:
-        # Check allclose() here for uniformity with _create_classifier_from_bddl
+    def _holding_classifier(self, state: State,\
+        objs: Sequence[Object]) -> bool:
+        # Check allclose() here for uniformity with
+        # _create_classifier_from_bddl
         assert state.allclose(self._current_ig_state_to_state())
         assert len(objs) == 1
         grasped_objs = self._get_grasped_objects(state)
         return objs[0] in grasped_objs
 
-    def _nextto_nothing_classifier(self, state: State,
-                                   objs: Sequence[Object]) -> bool:
-        # Check allclose() here for uniformity with _create_classifier_from_bddl
+    def _nextto_nothing_classifier(self, state: State,\
```

Comment:
Why do we still need this, now that we renamed nextto->reachable? What is the difference in the implementation of _reachable_nothing_classifier vs this one?

Reply:
good point; this is completely redundant and can be removed!